crates/iceberg/src/spec/schema.rs

```diff
@@ -642,6 +644,204 @@ impl SchemaVisitor for IndexByName {
     }
 }
 
+struct PruneColumn {
+    selected: HashSet<i32>,
+    select_full_types: bool,
+}
+
+/// Visit a schema and returns only the fields selected by id set
+pub fn prune_columns(
+    schema: &Schema,
+    selected: HashSet<i32>,
+    select_full_types: bool,
+) -> Result<Option<Type>> {
```

Comment:
Actually I use `Result<Option<Type>>` because in Java implementation there is precondition check which will trigger Exception.
```java
Preconditions.checkArgument(
            !field.type().isNestedType(),
            "Cannot explicitly project List or Map types, %s:%s of type %s was selected",
            field.fieldId(),
            field.name(),
            field.type());
```
I want to do the same thing in Rust but now I think maybe `panic!` is better like
```rust
panic!(
        "Cannot explicitly project List or Map types, {}:{} of type {} was selected",
        field.field_id(),
        field.name(),
        field.type()
);
```
Any suggestions?

Reply:
> I want to do the same thing in Rust but now I think maybe panic! is better like

We should avoid panic as much as possible since it cause an application crash. Also the error maybe caused by user input, which is an error and we should not panic.

In java's implementation, it tries to return empty struct or schema:
1. https://github.com/apache/iceberg/blob/684f7a767c2c216a402b60b73d2d55ef605921a0/api/src/main/java/org/apache/iceberg/types/TypeUtil.java#L74
2. https://github.com/apache/iceberg/blob/684f7a767c2c216a402b60b73d2d55ef605921a0/api/src/main/java/org/apache/iceberg/types/TypeUtil.java#L74

Reply:
OH I see. I misunderstood at first. Thanks for your explanation!!!